scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -26,3 +32,52 @@ impl SharedLibraryProvider for Package {
             .into_child(lib_name)
     }
 }
+
+pub fn compile_unit(unit: CompilationUnit, ws: &Workspace<'_>) -> Result<()> {
+    let main_package = unit.components.first().unwrap().package.clone();
+
+    let mut cmd = CargoCommand::builder()
+        .current_dir(main_package.root().to_path_buf())
+        .target_dir(
+            main_package
+                .target_path(ws.config())
+                .path_unchecked()
+                .to_path_buf(),
+        )
+        .build();
+
+    {
+        let _ = trace_span!("compile_proc_macro").enter();
+        let status = cmd
+            .status()
+            .with_context(|| format!("Failed to execute {:?}", cmd))?;
+        if !status.success() {
+            return Err(anyhow::anyhow!(
+                "Failed to compile procedural macro plugin: {:?}",
+                cmd
+            ));
+        }
+    }
+
+    Ok(())
+}
+
+#[derive(TypedBuilder)]
+#[builder(build_method(into = Command))]
+struct CargoCommand {
+    current_dir: Utf8PathBuf,
+    target_dir: Utf8PathBuf,
+}
+
+impl From<CargoCommand> for Command {
+    fn from(args: CargoCommand) -> Self {
+        let mut cmd = Command::new("cargo");
+        cmd.stdout(Stdio::inherit());
+        cmd.stderr(Stdio::inherit());
```

Comment:
I'm not sure. I think it'd be better if we use the stderr redirection mechanism that we're using for `git` calls. Stdout is fine.

Reply:
Note, that in Cargo all human addressed communication is on stderr - e.g. `Compiling scarb v2.5.3 (/[..]/scarb/scarb)` etc.

Reply:
Ok, so shouldn't we redirect this to stdout then to match what Scarb does?

Reply:
Hmm, to be honest I don't really see a point? Imho just passing both streams gives the most flexibility and the cleanest output 🤔 Would you mind to elaborate?